### PR TITLE
update 12/10/22

### PR DESCRIPTION
**Pull request by tientv34**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
-                            fontWeight: FontWeight.w700,
-                          ),
+    final screenSize = MediaQuery.of(context).size;
+    return ScaffoldBase(
+      backgroundColorScaffold: const Color(0xffF5F5F5),
+      body: Stack(
+        children: [
+          Center(
+            child: SingleChildScrollView(
+              padding:  EdgeInsets.only(right: 16,left: 16,top: screenSize.height*0.1),
```
Comment (minh-dai) [inline lib/main.dart:72]: please create a new class for EdgeInsets const

File: lib/main.dart [hunk 2]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: [
+                  Column(
+                    children:  [
+                      //Logo Images
+                      const IconWidget(),
+                      const SizedBox(height: 24),
+                      // Text ログイン
+                      Text(
+                        'ログイン',
```
Comment (minh-dai) [inline lib/main.dart:83]: please create a new class for language, because the app will be supporting multiple language

File: lib/main.dart [hunk 3]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
+                        style: AppTextStyle(
+                          color: ColorApp.color263645,
+                          letterSpacing: 0.35,
+                          fontSize: 22,
```
Comment (minh-dai) [inline lib/main.dart:87]: please create a new class for fontSize

File: lib/main.dart [hunk 4]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
-                      textError: errorName ? 'Error Text' : '',
-                      colorError: const Color(0xffDD052C),
-                    ),
-                    //Text ドメイン
-                    const SizedBox(
-                      height: 52,
+                      // textError: errorName ? 'Error Text' : '',
+                      validator: (name){
+                        if(name == null){
+                          return 'Error Text';
```
Comment (minh-dai) [inline lib/main.dart:115]: a same

File: lib/main.dart [hunk 5]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
                       controller: emailTextEditingController,
-                      textError: errorEmail ? 'Error Text' : '',
-                      colorError: const Color(0xffDD052C),
-                    ),
-                    const SizedBox(
-                      height: 52,
+                      //
+                      validator: (email){
+                        if(email == null || !validateEmail(email)!){
+                          return 'Error Text';
```
Comment (minh-dai) [inline lib/main.dart:139]: a same

File: lib/main.dart [hunk 6]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
-                              color: errorPassword
-                                  ? const Color(0xffDD052C)
-                                  : Colors.black.withOpacity(0.12)),
-                          borderRadius: BorderRadius.circular(4)),
                       onChanged: onChangePassword,
                       controller: passwordTextEditingController,
                       obscureText: isShowPass,
+                      validator: (password){
+                        if(!validatePassword(password!)!){
+                          return 'Error Text';
```
Comment (minh-dai) [inline lib/main.dart:162]: a same

File: lib/main.dart [hunk 7]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
+                  ),
+                  const SizedBox(
+                    height: 58,
+                  ),
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      Checkbox(
+                        shape: RoundedRectangleBorder(
+                            borderRadius: BorderRadius.circular(4)),
```
Comment (minh-dai) [inline lib/main.dart:184]: please create a new class for BorderRadius

File: lib/main.dart [hunk 8]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
+                        checkColor: ColorApp.white,
+                        activeColor: const Color(0xff008A7C),
+                        value: isCheck,
+                        onChanged: (bool? value) {
+                          setState(() {
```
Comment (minh-dai) [inline lib/main.dart:189]: please create a new function for this code. don't use logic code into view

File: lib/main.dart [hunk 9]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
+                        },
+                      ),
+                      const Padding(
+                        padding: EdgeInsets.only(left: 11),
+                        child: Text(
+                          'ログイン情報を保存する',
+                          style: AppTextStyle(
+                              fontFamily: 'fonts/sFProText.ttf',
+                              fontSize: 13,
+                              color: Color(0xff263645),
```
Comment (minh-dai) [inline lib/main.dart:201]: add color into Color class

File: lib/main.dart [hunk 10]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
+                        activeColor: const Color(0xff008A7C),
+                        value: isCheck,
+                        onChanged: (bool? value) {
+                          setState(() {
+                            isCheck = value!;
+                          });
+                        },
+                      ),
+                      const Padding(
+                        padding: EdgeInsets.only(left: 11),
```
Comment (minh-dai) [inline lib/main.dart:195]: a same, create a new class for EdgeInsets

File: lib/main.dart [hunk 11]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
-                                  color: Color(0xffFFFFFF),
-                                )),
-                          ),
+                        child: const Center(
+                          child: Text('ログイン',
+                              style: AppTextStyle(
+                                fontWeight: FontWeight.w600,
+                                fontSize: 17,
+                                letterSpacing: -0.408,
+                                color: Color(0xffFFFFFF),
```
Comment (minh-dai) [inline lib/main.dart:227]: a same

File: lib/main.dart [hunk 12]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
-                                  fontSize: 17,
-                                  letterSpacing: -0.408,
-                                  color: Color(0xffFFFFFF),
-                                )),
-                          ),
+                        child: const Center(
+                          child: Text('ログイン',
+                              style: AppTextStyle(
+                                fontWeight: FontWeight.w600,
+                                fontSize: 17,
```
Comment (minh-dai) [inline lib/main.dart:225]: a same

File: lib/main.dart [hunk 13]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
+                  const SizedBox(
+                    height: 24,
+                  ),
+                   Padding(
+                    padding: EdgeInsets.only(left: 16,right: 16, bottom: screenSize.height*0.08),
+                    child: const Text(
+                      'パスワードを忘れた方はこちら',
+                      textAlign: TextAlign.center,
+                      style: AppTextStyle(
+                        color: Color(0xff008A7C),
```
Comment (minh-dai) [inline lib/main.dart:242]: a same

File: lib/main.dart [hunk 14]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
-                          fontWeight: FontWeight.w500,
-                          fontSize: 15,
-                          letterSpacing: -0.24,
-                        ),
+                  ),
+                  const SizedBox(
+                    height: 24,
+                  ),
+                   Padding(
+                    padding: EdgeInsets.only(left: 16,right: 16, bottom: screenSize.height*0.08),
```
Comment (minh-dai) [inline lib/main.dart:237]: a same

File: lib/main.dart [hunk 15]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
+                    child: const Text(
+                      'パスワードを忘れた方はこちら',
+                      textAlign: TextAlign.center,
+                      style: AppTextStyle(
+                        color: Color(0xff008A7C),
+                        fontWeight: FontWeight.w500,
+                        fontSize: 15,
```
Comment (minh-dai) [inline lib/main.dart:244]: a same

**Review by minh-dai — CHANGES_REQUESTED** [review]

File: lib/main.dart (hunk 2 above)
Comment (tientv34) [reply lib/main.dart:83]: done

File: lib/main.dart (hunk 3 above)
Comment (tientv34) [reply lib/main.dart:87]: done

File: lib/main.dart (hunk 4 above)
Comment (tientv34) [reply lib/main.dart:115]: done

File: lib/main.dart (hunk 5 above)
Comment (tientv34) [reply lib/main.dart:139]: done

File: lib/main.dart (hunk 12 above)
Comment (tientv34) [reply lib/main.dart:225]: done

File: lib/main.dart (hunk 13 above)
Comment (tientv34) [reply lib/main.dart:242]: done

File: lib/main.dart (hunk 15 above)
Comment (tientv34) [reply lib/main.dart:244]: done

File: lib/main.dart [hunk 16]
```diff
@@ -58,206 +62,214 @@ class _MyHomePageState extends State<MyHomePage> {
...
+            Positioned.fill(
+              child: GestureDetector(
+                behavior: HitTestBehavior.translucent,
+                child: Container(
+                  alignment: Alignment.center,
+                  color: Colors.black.withOpacity(0.3),
+                  child: const SizedBox(
+                      height: 50,
+                      width: 50,
+                      child: CircularProgressIndicator(
```
[the remark this answers is not in the document]
Comment (tientv34) [reply lib/main.dart:263]: done

**Pull request closed without merging**